controllers/che/checluster_controller.go

```diff
@@ -251,16 +250,11 @@ func (r *CheClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	// Fetch the CheCluster instance
-	checluster, err := r.GetCR(req)
-
-	if err != nil {
-		if errors.IsNotFound(err) {
-			r.Log.Info("CheCluster Custom Resource not found.")
-			// Request object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
-			// Return and don't requeue
-			return ctrl.Result{}, nil
-		}
+	checluster, err := deploy.FindCheClusterCRInNamespace(r.nonCachedClient, req.NamespacedName.Namespace)
+	if checluster == nil {
+		r.Log.Info("CheCluster Custom Resource not found.")
+		return ctrl.Result{}, nil
+	} else if err != nil {
```

Comment:
We might want to add a secondary check here -- the previous functionality would get the specific CheCluster that triggered the reconcile (i.e. only return 0 or 1 CheClusters). Now it's possible for this section to return an error because there are 2+ CheClusters in the same namespace. If this occurs, the operator will enter a tight loop of 

> (queue reconcile for CheCluster A) -> (find two CheClusters, return error) -> (queue reconcile for CheCluster A).

If there are multiple CheClusters in a namespace, they should get some sort of error state and stop queueing reconciles until one of them is deleted.

Reply:
Thank you for your comment.
che-operator doesn't support more than one checluster and the webhook [1] prevents creating it.
So, it is not even possible to bump into the such an error, that's why I don't think it worth changing anything.

[1] https://github.com/eclipse-che/che-operator/blob/a4e19ead9d324dce09f5ca400a5cb7eff3abd253/api/v2/checluster_webhook.go#L86

Reply:
Makes sense :+1: